runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -122,11 +131,53 @@ decl_module! {
 			// And track that we've finished processing the inherent for this block.
 			Included::set(Some(()));
 
-			Ok(())
+			Ok(Some(
+				MINIMAL_INCLUSION_INHERENT_WEIGHT +
+				(backed_candidates_len * BACKED_CANDIDATE_WEIGHT)
+			).into())
 		}
 	}
 }
 
+/// Limit the number of backed candidates processed in order to stay within block weight limits.
+///
+/// Use a configured assumption about the weight required to process a backed candidate and the
+/// current block weight as of the execution of this function to ensure that we don't overload
+/// the block with candidate processing.
+///
+/// If the backed candidates exceed the available block weight remaining, then skips all of them.
+/// This is somewhat less desirable than attempting to fit some of them, but is more fair in the
+/// even that we can't trust the provisioner to provide a fair / random ordering of candidates.
+fn limit_backed_candidates<T: Config>(
+	backed_candidates: Vec<BackedCandidate<T::Hash>>,
+) -> Vec<BackedCandidate<T::Hash>> {
+	let block_weight_remaining = <T as frame_system::Config>::BlockWeights::get().max_block
+		.saturating_sub(frame_system::Module::<T>::block_weight().total());
+
+	if backed_candidates.len() as Weight * BACKED_CANDIDATE_WEIGHT > block_weight_remaining  {
```

Comment:
block_weight_remaining already include `INCLUSION_INHERENT_CLAIMED_WEIGHT` here. it has been added to block weight with the extension check_weight in frame_system.

So you just need to check if `<T as frame_system::Config>::BlockWeights::get().max_block < frame_system::Module::<T>::block_weight().total()`

Reply:
Yup: 84afca72.